docs/plugins/repository-s3.asciidoc

```diff
@@ -12,98 +12,112 @@ include::install_remove.asciidoc[]
-elasticsearch config using `cloud.aws.access_key` and `cloud.aws.secret_key` or
-if you wish to set credentials specifically for s3 `cloud.aws.s3.access_key` and `cloud.aws.s3.secret_key`:
+credentials for authentication. The only mandatory setting is the bucket name:
 
-[source,yaml]
+[source,js]
 ----
-cloud:
-    aws:
-        access_key: AKVAIQBF2RECL7FJWGJQ
-        secret_key: vExyMThREXeRMm/b/LRzEB8jWwvzQeXgjqMX+6br
+PUT _snapshot/my_s3_repository
+{
+  "type": "s3",
+  "settings": {
+    "bucket": "my_bucket"
+  }
+}
 ----
+// CONSOLE
+// TEST[skip:we don't have gcs setup while testing this]
+
 
-[[repository-s3-usage-security]]
-===== Transport security
+[[repository-s3-client]]
+==== Client Settings
 
-By default this plugin uses HTTPS for all API calls to AWS endpoints. If you wish to configure HTTP you can set
-`cloud.aws.protocol` in the elasticsearch config. You can optionally override this setting per individual service
-via: `cloud.aws.ec2.protocol` or `cloud.aws.s3.protocol`.
+The client used to connect to S3 has a number of settings available. Client settings names are of
+the form `s3.client.CLIENT_NAME.SETTING_NAME` and specified inside `elasticsearch.yml`.  The
+default client name is `default`, but can be customized with the repository setting `client`.
+For example:
 
-[source,yaml]
+[source,js]
 ----
-cloud:
-    aws:
-        protocol: https
-        s3:
-            protocol: http
-        ec2:
-            protocol: https
+PUT _snapshot/my_s3_repository
+{
+  "type": "s3",
+  "settings": {
+    "bucket": "my_bucket",
+    "client": "my_alternate_client"
+  }
+}
 ----
+// CONSOLE
+// TEST[skip:we don't have gcs setup while testing this]
 
-In addition, a proxy can be configured with the `proxy.host`, `proxy.port`, `proxy.username` and `proxy.password` settings
-(note that protocol can be `http` or `https`):
+The following settings name be specified for a client.  Some settings are sensitive and must be
+stored in the {ref}/secure-settings.html[elasticsearch keystore], and are marked as `Secure`.
 
-[source,yaml]
-----
-cloud:
-    aws:
-        protocol: https
-        proxy:
-            host: proxy1.company.com
-            port: 8083
-            username: myself
-            password: theBestPasswordEver!
-----
+`access_key`::
 
-You can also set different proxies for `ec2` and `s3`:
+    An s3 access key. Must be specified with `secret_key`. (Secure)
 
-[source,yaml]
-----
-cloud:
-    aws:
-        s3:
-            proxy:
-                host: proxy1.company.com
-                port: 8083
-                username: myself1
-                password: theBestPasswordEver1!
-        ec2:
-            proxy:
-                host: proxy2.company.com
-                port: 8083
-                username: myself2
-                password: theBestPasswordEver2!
-----
+`secret_key`::
 
-===== Read timeout
+    An s3 secret key. Must be specified with `access_key`. (Secure)
```

Comment:
What is there is correct. When configuring a secret_key, an access_key must also be configured.